satpy/etc/readers/cmsaf-claas2_l2_nc.yaml

```diff
@@ -0,0 +1,21 @@
+reader:
+  description: >
+    Reader for Spinning Enhanced Visible and Infrared Imager (SEVIRI) L2 data as produced by
+    the Climate Monitoring
+    Satellite Application Facility (CMSAF) in its CLoud property dAtAset using SEVIRI (CLAAS) dataset,
+    doi:10.5676/EUM_SAF_CM/CLAAS/V002.
+    Information on CMSAF and its products can be found at https://www.cmsaf.eu/.
+  name: cmsaf-claas2_l2_nc
+  long_name: CMSAF CLAAS V2 data for SEVIRI-derived cloud products
+  sensors: [seviri]
+  doi: doi:10.5676/EUM_SAF_CM/CLAAS/V002.
+  default_channels: []
+  reader: !!python/name:satpy.readers.yaml_reader.FileYAMLReader
+
+# CMSAF naming convention:
+#   https://www.cmsaf.eu/EN/Products/NamingConvention/Naming_Convention_node.html
+
+file_types:
+    cmsaf-claas-v2:
+        file_reader: !!python/name:satpy.readers.cmsaf_claas_v2.Claasv2
+        file_patterns: ['{product:3s}in{start_time:%Y%m%d%H%M}{version:>03d}05SVMSG01MD.nc']
```

Comment:
I think the version number should be specified explicitly to prevent confusion with older/newer CLAAS datasets whose data format is or will be a bit different. For the CLAAS-2 dataset the version is fixed to `003` and for the corresponding ICDR "continuation" the pattern is `4??` (400 & 410 currently). I know this is hard to explain to users. We're working on an improvement :slightly_smiling_face:

Reply:
I thought the 2 in `CLAAS-2` was the version number, I'm confused.

Reply:
Oh, or do you mean in the `file_patterns` and the `file_types`?  The reader could still be named what it is, just opening the possibility for other classes to be added for reading files with other 3-digit versions?

Reply:
Yes, I was talking about the `{version:>03d}` part. But I think you're right, it's better to keep it generic.

Reply:
But I'm noticing another thing here: The CLAAS-2 extension has `SVMSGE1MD` instead of `SVMSG01MD` :see_no_evil: That's horrible... So you could do `SVMSG{proc_level}MD.nc`

Reply:
I've added `proc_level`.